app/importers/import_task.rb

```diff
@@ -1,38 +1,45 @@
 class ImportTask
-  def initialize(options:)
-    @options = options
+  # If passed invalid arguments, validete and fail in the constructor
+  def initialize(options = {})
+    @options = options.symbolize_keys
 
-    # options["school"] is an optional filter; imports all schools if nil
-    @school = @options.fetch("school", nil)
+    # which schools to include
+    @school_local_ids = validate_school_local_ids!(@options[:school_local_ids])
 
-    # options["source"] describes which external data sources to import from
-    @source = @options.fetch("source", ["x2", "star"])
+    # which importers to run
+    @importer_descriptions = validate_importer_keys!(@options[:importer_keys])
 
-    # options["only_recent_attendance"]
-    @only_recent_attendance = @options.fetch("only_recent_attendance", false)
+    # only recent attendance data, or all of it
+    raise 'missing only_recent_attendance' unless @options.has_key?(:only_recent_attendance)
+    @only_recent_attendance = @options[:only_recent_attendance]
 
-    @log = Rails.env.test? ? LogHelper::Redirect.instance.file : STDOUT
+    # log differently based on env
+    @log = pick_log_based_on_env(options)
   end
 
   def connect_transform_import
     begin
+      log('Setting up logging...')
       @record = create_import_record
+      log('Done logging.')
+
+      log('Running inital report...')
       @report = create_report
-      log('Starting validation...')
-      validate_district_option
-      seed_schools_if_needed
-      validate_school_options
```

Comment:
Validation is moved up to the constructor so invalid args fail before any work can be done, and seeding is removed here.